### PR TITLE
Add metadata.messageType to protobuf payloads

### PR DESCRIPTION
**Pull request by lorensr**
Matches this PR in the Java SDK: https://github.com/temporalio/sdk-java/pull/942

I'm new to Go—feedback welcome! 🤗

### CODE REVIEW

File: converter/proto_helpers.go
Comment (lorensr) [inline converter/proto_helpers.go:43]: I'm guessing this should be using the `log` package? Perhaps a `Logger` instance that the user configures when setting up the Client or Worker? I'm wondering how to get that instance.

File: CONTRIBUTING.md
Comment (cretz) [inline CONTRIBUTING.md:17]: Code code block is not properly nested under the bullet (see https://gist.github.com/clintel/1155906)

File: converter/payload_converter.go
Comment (cretz) [inline converter/payload_converter.go:45]: I don't think this needs to be exported

File: converter/proto_helpers.go
Comment (cretz) [reply converter/proto_helpers.go:43]: There's no available log here. I am ok not checking this. Unfortunately unlike Java we have to pass around log instances.

File: converter/proto_helpers.go
Comment (cretz) [inline converter/proto_helpers.go:35]: I don't think this stuff needs to be public API

File: converter/proto_json_payload_converter.go
Comment (cretz) [inline converter/proto_json_payload_converter.go:48]: Nobody can set this value because it is not exported

File: converter/proto_json_payload_converter.go
Comment (cretz) [inline converter/proto_json_payload_converter.go:47]: All exported items should have documentation that starts with its name, e.g. `// ProtoJSONPayloadConverterOptions represents options for `NewProtoJSONPayloadConverterWithOptions`.

File: converter/payload_converter.go
Comment (lorensr) [reply converter/payload_converter.go:45]: Ah, thanks. I was using capitals for using it outside of the file, but it's for outside of the package.

**Review by cretz — APPROVED** [review]
Looks good! Seems CI failed in a spurious way, I have restarted it. Once complete successfully, feel free to merge.

**Comment by lorensr** [comment]
Thanks! I can't merge (don't have write access) or view buildkite (first says "login", then says "page not found", I assume due to permissions). `make check` and my `make test` with `MallocNanoZone=0` both exit with code 0.

**Comment by cretz** [comment]
:+1: I will look into the build errors then do the merge

**Pull request merged**
